src/Motor.Extensions.Diagnostics.Telemetry/TelemetryHostBuilderExtensions.cs

```diff
@@ -0,0 +1,154 @@
+                            .AddAspNetCoreInstrumentation(options =>
+                            {
+                                options.Filter = AspNetCoreFilter(openTelemetryOptions);
+                            })
+                            .AddHttpClientInstrumentation(options =>
+                            {
+                                options.Enrich = httpClientInstrumentationOptions.Enrich;
+                                options.Filter = httpClientInstrumentationOptions.Filter;
+                                options.SetHttpFlavor = httpClientInstrumentationOptions.SetHttpFlavor;
+                            })
+                            .AddSource(OpenTelemetryOptions.DefaultActivitySourceName)
+                            .AddSource(openTelemetryOptions.Sources.ToArray())
+                            .SetResourceBuilder(ResourceBuilder.CreateDefault()
+                                .AddService(applicationNameService.GetFullName(),
+                                    serviceVersion: applicationNameService.GetVersion())
+                                .AddAttributes(new[]
+                                {
+                                    new KeyValuePair<string, object>(AttributeMotorNetEnvironment,
+                                        hostContext.HostingEnvironment.EnvironmentName.ToLower()),
+                                    new KeyValuePair<string, object>(AttributeMotorNetLibraryVersion,
+                                        applicationNameService.GetLibVersion()),
+                                }))
+                            .SetMotorSampler(openTelemetryOptions);
+                        if (otelSection != null)
+                        {
+                            builder.AddOtlpExporter(logger, provider);
+                        }
+                        else
+                        {
+                            builder.AddJaegerExporter(logger, provider);
+                        }
+                    });
+                });
+            return hostBuilder;
+        }
+
+        private static Func<HttpContext, bool> AspNetCoreFilter(OpenTelemetryOptions openTelemetryOptions)
+        {
+            return req => !openTelemetryOptions.FilterOutTelemetryRequests ||
+                          !req.Request.Path.StartsWithSegments("/metrics") &&
+                          !req.Request.Path.StartsWithSegments("/health");
+        }
+
+        private static TracerProviderBuilder SetMotorSampler(this TracerProviderBuilder builder,
+            OpenTelemetryOptions options)
+        {
+            Sampler sampler = Math.Abs(options.SamplingProbability - 1.0) < 0.0001
+                ? new AlwaysOnSampler()
+                : new TraceIdRatioBasedSampler(options.SamplingProbability);
+
+            builder.SetSampler(new ParentBasedSampler(sampler));
+            return builder;
+        }
+
+        private static void AddJaegerExporter(this TracerProviderBuilder builder, ILogger logger,
```

Comment:
nit: even with having this in the new package isn't also kinda obsolete already?

Reply:
kinda obsolete